SIPS/sip-63.md

```diff
@@ -28,21 +28,80 @@ There is currently no way to track trading fees paid on-chain, so it is not poss
+
+**Interface**
+
+```
+pragma solidity >=0.4.24;
+
+interface ITradingIncentive {
+  // Views
+  function rewards(uint periodID) external view; 
+  
+  function rewardForPeriods(uint[] periodIDs) external view;
+  
+  // Mutative Functions
+  function recordExchangeFee(uint amount, address account) external;
+  
+  function claimReward(uint periodID) external;
+  
+  function claimRewardForPeriods(uint[] periodIDs) external;
+  
+  // owner only
+  function notifyRewardAmount(uint reward) external;
+}
+```
+
+**Exchanger contract**
+
+Remit exchange fees generated by an account to the `Trading incentive` contract, storing the total fees for each account in the contract. 
+
+A function to allow trading incentives to be recorded against any `address` instead of the `msg.sender` of the `synthetix.exchange()` event. The `address` will need to be the one claiming the rewards from the `trading incentive` contract.    
+
 ### Test Cases
 <!--Test cases for an implementation are mandatory for SIPs but can be included with the implementation..-->
-Test cases for an implementation are mandatory for SIPs but can be included with the implementation.
+- When an exchange is made, the exchange fee in sUSD and the address the exchange `from` gets stored in the trading incentives contract and the `totalFeesInPeriod` is increased by the exchange fee amount. 
+- When an exchange is made, the exchange fee is added to the `totalExchangeFees` for the account, a value accumulated for the period until it is closed. 
+- When an exchange is made, if the account last made a trade in a previous period, it will store their `totalFeesInPeriod` for the previous period and reset the value for the current period.
+- When an exchange is made, if an address is passed in to record the trading incentives for, then it will store the total fees against that address.
+- When notifyRewards is called, if the SNX balance is already transferred to the contract, then it passes. The transaction will record the `totalFeesInPeriod` for the current period and reset it. It will also increment the `periodID`.
+
+**Reward Claiming**
+
+- When a user claim their rewards, their reward will be calculated as a proportion of the fees they generated against the total fees of that period and the period is marked claimed for the user.
+- When a user claim their rewards, if the claim is successful, it will record the final `totalFeesInPeriod` for the last  reward period if the period is closed, and reset their current `totalFeesInPeriod`. 
+- When a user claim their rewards, if the claim is successful, it will increase the `totalRewardsClaimed` amount which ensures the contract has enough SNX to pay the rewards (on notifyRewards it will check the contract balance less `totalRewardsClaimed`.   
+- When a user claim their rewards, if the period they provide has no fees generated, then it will fail.
```

Comment:
[np] This is better written as `Given When Then` using cucumber BDD syntax. So `Given a period that has no fees for a user, when that user clams for that period, then the transaction fails` or some such